src/overlays/actors/ovl_Obj_Lift/z_obj_lift.c

```diff
@@ -30,29 +32,182 @@ const ActorInit Obj_Lift_InitVars = {
+typedef struct {
+    s16 x;
+    s16 z;
+} unk_temp_name;
+unk_temp_name D_80B969D0[] = { { 120, -120 }, { 120, 0 },     { 120, 120 }, { 0, -120 },  { 0, 0 },
+                               { 0, 120 },    { -120, -120 }, { -120, 0 },  { -120, 120 } };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Lift/func_80B96560.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Lift/func_80B9664C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Lift/func_80B96678.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Lift/func_80B967C0.s")
+// sInitChain
+InitChainEntry D_80B969F4[] = {
+    ICHAIN_F32_DIV1000(gravity, -600, ICHAIN_CONTINUE),   ICHAIN_F32_DIV1000(minVelocityY, -15000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 2000, ICHAIN_CONTINUE), ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 2000, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Lift/func_80B96840.s")
+f32 D_80B96A08[] = { 0.1f, 0.05f };
+f32 D_80B96A10[] = { -18.0f, -9.0f, 0, 0 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Lift/ObjLift_Update.s")
+extern UNK_TYPE D_06000140;
+extern UNK_TYPE D_060003D0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Lift/ObjLift_Draw.s")
+void func_80B960D0(ObjLift* this, s32 arg1) {
+    this->actionFunc = arg1;
+}
+
+void func_80B960D8(ObjLift* this, GlobalContext* globalCtx, void* colHeader, DynaPolyMoveFlag flags) {
+    s32 pad;
+    CollisionHeader* sp30 = NULL;
+    s32 pad2;
+
+    DynaPolyActor_Init(this, flags);
+    CollisionHeader_GetVirtual(colHeader, &sp30);
+    this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, sp30);
+    if (this->dyna.bgId == 0x32) {
+        osSyncPrintf((const char*)"Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_obj_lift.c",
+                     0xBC, this->dyna.actor.id, this->dyna.actor.params);
+    }
+}
+
+void func_80B96160(ObjLift* this, GlobalContext* globalCtx) {
+    Vec3f pos;
+    Vec3f velocity;
+    Vec3f* temp_s3;
+    s32 pad0;
+    s32 i;
+
+    temp_s3 = &this->dyna.actor.world.pos;
+    for (i = 0; i < 9; i++) {
+        pos.x = D_80B969D0[i].x * this->dyna.actor.scale.x + temp_s3->x;
+        pos.y = temp_s3->y;
+        pos.z = D_80B969D0[i].z * this->dyna.actor.scale.z + temp_s3->z;
+        velocity.x = D_80B969D0[i].x * this->dyna.actor.scale.x * 0.8f;
+        velocity.y = Rand_ZeroOne() * 10.0f + 6.0f;
+        velocity.z = D_80B969D0[i].z * this->dyna.actor.scale.z * 0.8f;
+        EffectSsKakera_Spawn(globalCtx, &pos, &velocity, temp_s3, -0x100, (Rand_ZeroOne() < 0.5f) ? 0x40 : 0x20, 0xF,
+                             0xF, 0, (Rand_ZeroOne() * 50.0f + 50.0f) * this->dyna.actor.scale.x, 0, 0x20, 0x32, -1,
+                             0x11D, &D_06000140);
```

Comment:
```suggestion
        EffectSsKakera_Spawn(globalCtx, &pos, &velocity, temp_s3, -0x100, (Rand_ZeroOne() < 0.5f) ? 0x40 : 0x20, 0xF,
                             0xF, 0, (Rand_ZeroOne() * 50.0f + 50.0f) * this->dyna.actor.scale.x, 0, 0x20, 0x32, -1,
                             OBJECT_D_LIFT, &D_06000140);
```
There are other things that can be tweaked here that I don't know offhand but Fig will probably point out